src/pages/profile-page/components/shipping-addresses.tsx

```diff
@@ -3,17 +3,29 @@ import { messages } from '../../../sources/messages';
 import { userStore } from '../../../store/user-store';
 import { AddressCard } from './address-card/address-card';
 import styles from '../profile-page.module.css';
+import { useEffect, useState } from 'react';
 
 export const ShippingAddresses = observer(() => {
   const addresses = userStore.shippingAddresses;
-
+  const [defaultAddressID, setDefaultAddressID] = useState<string>();
+  useEffect(() => {
+    setDefaultAddressID(userStore.user?.defaultShippingAddressId || '');
+  }, [userStore.user]);
```

Comment:
```suggestion
  const [defaultAddressID, setDefaultAddressID] = useState<string>(
    userStore.user?.defaultShippingAddressId || ''
  );
```
а чего просто так нельзя?

Reply:
Здесь суть в том что юзер появляется позже чем компонент рендерится

Reply:
и мы в useState запихиваем id как initial value и он почему-то не актулизируется
но при этом когда в обычную контанту записываешь значение из стора то она актуализируется